### ISSUE TITLE
[temp.deduct.general]/note-3 The note is a bit misleading

### ISSUE THREAD

**Issue by xmh0511** [cplusplus/CWG#275]
Full name of submitter (unless configured in github; will be published with the issue): Jim X

[temp.deduct.general]/note-3 says:
> Overload resolution will check the other parameters, including **parameters with dependent types in which no template parameters participate in template argument deduction**

Anyway, such a parameter should make the deduction fail, this means, overload resolution won't further check it. The [temp.deduct.type] p19 says:
> Template parameters do not participate in template argument deduction if they are used only in non-deduced contexts.  

[temp.deduct.type] p4 says:
> If a template parameter is used only in non-deduced contexts and is not explicitly specified, **template argument deduction fails**.  

For such a template parameter, either it is explicitly specified or obtained from the default argument. Otherwise, it is a deduction failure.

**Comment by frederick-vs-ja** [comment]
IIUC the note is talking about such cases:
```C++
template<class T>
void f(T, std::type_identity_t<T>);

int main()
{
    f(42, 0.0);
}
```

In this example, `T` participates in template argument deduction in the first function parameter, but not in the second one.

I think the note can be delicately reworded, but this should be an editorial issue.

**Comment by jensmaurer** [comment]
Yes.  Note the "only"; a template parameter may be deducible from one parameter but appear only in non-deduced contexts in another parameter. The note highlights the checking in overload resolution for the other parameter.

I'm not seeing a defect here.

**Comment by jensmaurer** [comment]
This is a note. It doesn't need to be exhaustive. Obviously, overload resolution won't check anything if the process bails out before (e.g. due to deduction failure).

**Comment by xmh0511** [comment]
> This is a note. It doesn't need to be exhaustive. Obviously, overload resolution won't check anything if the process bails out before (e.g. due to deduction failure).

I think we should remove "parameters with dependent types in which no template parameters participate in template argument deduction" from the note. The intended meaning should be:
> parameters with dependent types where the template parameters are in the non-deduced contexts but their values can be obtained from other pairs. 

This is too long. As this is just a note, hence remove it to avoid further misleading is a trade-off way.
